nodeup/pkg/model/kubeapiserver.go

```diff
@@ -255,6 +256,41 @@ func (b *KubeAPIServerBuilder) buildPod() (*v1.Pod, error) {
 		}
 	}
 
+	if b.Cluster.Spec.EgressProxy != nil {
+		proxies := b.Cluster.Spec.EgressProxy
+		httpProxy := proxies.HTTPProxy
+		if httpProxy.Host != "" {
+			url := "http://"
+			if httpProxy.User != "" {
+				url += httpProxy.User
+				if httpProxy.Password != "" {
+					url += ":" + httpProxy.Password
+				}
+				url += "@"
+			}
+			url += httpProxy.Host + ":" + strconv.Itoa(httpProxy.Port)
+			container.Env = append(container.Env,
+				v1.EnvVar{
+					Name:  "http_proxy",
+					Value: url,
+				},
+				v1.EnvVar{
+					Name:  "https_proxy",
+					Value: url,
+				},
+				v1.EnvVar{
+					Name:  "ftp_proxy",
+					Value: url,
+				})
+		}
+		if proxies.ProxyExcludes != "" {
+			x := v1.EnvVar{
+				Name:  "no_proxy",
```

Comment:
My coworker noted that we had issues with the case variable names in our setup. I think it's okay in this spot but if you're having issues with `no_proxy` elsewhere try `NO_PROXY`.